### PR TITLE
Bit Manipulation 

### LINKED ISSUE #1835: Bit Manipulation
### Description

Description
what is bit manipulation 
types of bit operators 
uses and functioning.

### Domain

Competitive Programming

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by uditkatyal**
<hr>

## Description 📜

Bit manipulation explained
Types of bitwise operator
Proper explanation with images/gifs

Fixes #1835 

<hr>

## Type of change 📝

- [x] Doc#umentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] Competitive #Programming

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by atarax665** [comment]
The document should be renamed as ```bitwise_operators.md``` as it contains only the introduction of Bitwise operators.

File: Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# Bit Manipulation
+
+## Use of Bit Operators
+
+Bit operators perform operations much faster than arthmetic or logical operators
```
Comment (atarax665) [inline Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md:5]:
```suggestion
Bitwise operators perform operations much faster than arithmetic or logical operators.
```

File: Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
+# Bit Manipulation
```
Comment (atarax665) [inline Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md:1]:
```suggestion
# Bitwise Operators
```

File: Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+## Use of Bit Operators
+
+Bit operators perform operations much faster than arthmetic or logical operators
+
+## Types of Bit Operator
```
Comment (atarax665) [inline Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md:7]:
```suggestion
## Types of Bitwise Operator
```

File: Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+There are mainly 6 types of bit operators
```
Comment (atarax665) [inline Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md:9]:
```suggestion
There are mainly 6 types of bitwise operators
```

File: Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+   <br/><br>
+
+   ![Symbols representing Bitwise operator](https://miro.medium.com/max/540/1*lwKj-TpvToBxuq98LXII1A.png)
+
+<br/><br>
+
+## **_Bitwise AND_**
+
+The bitwise AND operator takes two arguments and performs AND operation on the operands bit by bit.
+The operation will only be 1 when both operands are 1 otherwise it remains 0.
```
Comment (atarax665) [inline Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md:27]:
```suggestion
The operation will be 1 only when both the bits are 1 otherwise it remains 0.
```

File: Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+![Table](https://i1.wp.com/techvidvan.com/tutorials/wp-content/uploads/sites/2/2019/12/implementation-of-and-operation-on-binary-digits.jpg?resize=341%2C272&ssl=1)
+
+<br />
+
+## **_Bitwise OR_**
+
+The bitwise OR operator takes two arguments and performs OR operation on each bit of the value of the operand. In the OR operation in binary, the resultant is 0 when both operands are 0 otherwise it is 1.
```
Comment (atarax665) [inline Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md:35]:
```suggestion
The bitwise OR operator takes two arguments and performs OR operations on each bit of the operands. In the OR operation, the resultant is 0 when i-th bit of both the operands is 0 otherwise the result is 1.
```

File: Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+![](https://i2.wp.com/techvidvan.com/tutorials/wp-content/uploads/sites/2/2019/12/bitwise-or-operations.jpg?resize=325%2C238&ssl=1)
+
+<br />
+
+## **_Bitwise XOR_**
+
+The bitwise Xor takes two operands and performs Xor operation on the binary digits. The Xor operation gives 0 as a result when both operands are the same and 1 when operands are different.
```
Comment (atarax665) [inline Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md:43]:
```suggestion
The bitwise XOR takes two operands and performs XOR operation on the binary digits. The Xor operation gives 0 as a result when i-th bit of both the operands is same and 1 when the bits are different.
```

File: Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+The left shift operator shifts the bits of the numbers by a specified number of places.
+
+`Example 1`:
+
+Let’s see how this works.
+
+**2<<2 = 8**
+
+2 is represented as ‘010’
```
Comment (atarax665) [inline Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md:59]:
```suggestion
2 is represented as ‘010’ in binary form.
```

File: Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+We shift the digits to _left up_ to two places. So, it will be ‘1000’ in binary and 8 in decimal.
```
Comment (atarax665) [inline Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md:61]:
```suggestion
We shift the digits to _left up_ to two places. So, it will become ‘1000’ in binary and 8 in decimal.
```

File: Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+Shifting 4 digits to the _left_ will make it ‘011 0000’ which is 48 in decimal.
+
+Shortcut for \*\*left shit operator
+
+<br />
+
+## **_Right Shift Operator_**
+
+This is similar to the left shift, but instead of shifting digits to the left we are shifting them to the right. In this process, some of the bits are lost.
```
Comment (atarax665) [inline Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md:77]:
```suggestion
This is similar to the left shift, but instead of shifting digits to the left, we shift them to the right. In this process, some of the bits get discarded.
```

File: Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+**20>>2 = 5**
+
+20 in binary is “10100”. We _right shift_ them up to 2 places so, we are now left with “101” which is 5.
+
+<br/>
+
+## **_Bitwise NOT_**
+
+This operator requires single input and does its negation
+If the input value is one it gets converted to zero
```
Comment (atarax665) [inline Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md:102]:
```suggestion
If the i-th bit is 1 it gets converted to 0 and vice-versa.
```

File: Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+
+**20>>2 = 5**
+
+20 in binary is “10100”. We _right shift_ them up to 2 places so, we are now left with “101” which is 5.
+
+<br/>
+
+## **_Bitwise NOT_**
+
+This operator requires single input and does its negation
```
Comment (atarax665) [inline Competitive_Programming/C++/Bit Manipulation/bit_manipulation.md:101]:
```suggestion
This operator requires single input and inverts all the bits of the given input.
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by uditkatyal** [comment]
I have changed accordingly thanks for the review 
Is it good to go for merge! @atarax665

**Comment by uditkatyal** [comment]
@atarax665  is it good to go for merge?

File: Open_Source/Section-4/4.14_Open_Mainframe_Project_Mentorship_Program.md [hunk 13]
```diff
@@ -0,0 +1,43 @@
+## All About Open Mainframe Project Mentorship Program
```
Comment (atarax665) [inline Open_Source/Section-4/4.14_Open_Mainframe_Project_Mentorship_Program.md:1]: This file doesn't belong to this PR. Please remove it.

File: Competitive_Programming/C++/Bit Manipulation/README.md [hunk 14]
```diff
@@ -0,0 +1 @@
+`Bit Manipulation`
```
Comment (atarax665) [inline Competitive_Programming/C++/Bit Manipulation/README.md:1]: Do not edit the README.md. We will be doing it later in a systematic order to maintain the consistency of the repository.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by uditkatyal** [comment]
@atarax665  I have made the changes to actual file and no change to readme 
I hope it is good to go now!

File: Open_Source/Section-4/4.14_Open_Mainframe_Project_Mentorship_Program.md (hunk 13 above)
Comment (atarax665) [reply Open_Source/Section-4/4.14_Open_Mainframe_Project_Mentorship_Program.md:1]: @uditkatyal Please remove this file.

**Review by atarax665 — COMMENTED** [review]
@uditkatyal Please remove `Open_Source/Section-4/4.14_Open_Mainframe_Project_Mentorship_Program.md` so that I can merge this PR.
